### PR TITLE
add missing copy on access conversion for GetItemSetter

### PR DESCRIPTION
**Pull request by MikeHolman**

### CODE REVIEW

**Comment by suwc** [comment]
:shipit:

**Comment by suwc** [comment]
LGTM

**Comment by akroshg** [comment]
>         virtual DescriptorFlags GetItemSetter(uint32 index, Var* setterValue, ScriptContext* requestContext) override

what about this case?


---
Refers to: lib/Runtime/Library/JavascriptArray.h:1113 in a9cb55a. [](commit_id = a9cb55ac121b4df344f539e7e0f776be798bc771, deletion_comment = False)

File: test/Array/CopyOnAccessArray_bugs.js [hunk 1]
```diff
@@ -129,6 +129,13 @@ var tests = [
         }
     },
     {
+        name: "Reflect.set",
+        body: function ()
+        {
+            assert.isTrue(Reflect.set([1950, 1960, 1970, 1980, 1990], "0", 1), "Should be able to set property on array");
```
Comment (akroshg) [inline test/Array/CopyOnAccessArray_bugs.js]: add the test case for float and mix floats and ints.

**Comment by MikeHolman** [comment]
@akroshg I think that case shouldn't happen, because you won't have a CopyOnAccessIntArray if you have NativeFloatArray vtable.

**Pull request merged**